astropy/stats/sigma_clipping.py

```diff
@@ -194,6 +209,23 @@ def perform_clip(_filtered_data, _kwargs):
         warnings.warn("Input data contains invalid values (NaNs or infs), "
                       "which were automatically masked.", AstropyUserWarning)
 
+    if HAS_SCIPY is True:
+        if masked is False:
+            if cenfunc is not np.ma.mean or stdfunc is not np.std:
```

Comment:
Sorry to be so late in the game, but I don't think this should just be a warning: the code cannot do what is requested, so it should raise an exception, not just put out a warning.

Alternatively, it should just use the slow path, as is now the case when `scipy` is not present.